src/OffensivePlant.java

```diff
@@ -1,18 +1,16 @@
-public abstract class OffensivePlant extends Plant {
-    public OffensivePlant(int hp, int sunPointCost, int abilityFrequency) {
-        super(hp, sunPointCost, abilityFrequency);
-    }
-
-    public Projectile generateProjectile() {
-		this.turnsUntilAbility--;
-
-		if (this.turnsUntilAbility == 0) {
-			this.turnsUntilAbility = this.abilityFrequency;
-			return new PeashooterProjectile();
-		}
-
-		return null;
-	}
-
-    public abstract String toString();
-}
\ No newline at end of file
+/* OffensivePlant is a subclass of Plants.
+ * This is the initial break down to separate damage dealing plants and sunflower.
+ * OffensivePlants shoots out a kinetic projectile and type component of firearm ammunition.*/
+public abstract class OffensivePlant extends Plants {
+    public OffensivePlant(int hp, int sunPointCost, int abilityFrequency) {
+        super(hp, sunPointCost, abilityFrequency);
+    }
+
+    // using "this" works in this case, but it should be "super" to prevent smelly code since we're calling from super class.
+    public Projectile generateProjectile() {
+    	// PeaShooter should shoot right away
+    	return new PeashooterProjectile(1, this.abilityFrequency);

```

Comment:
my mistake from previous push, fix it later so that it isn't a peashooter projectile, should keep the old code commented out because that logic will be reimplemented anyways so that it works for every offensive plant

Reply:
also the speed parameter is how fast the bullet travels, not its frequency, the frequency is how often it shoots